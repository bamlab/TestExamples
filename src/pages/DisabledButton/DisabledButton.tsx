import React, { useState } from 'react';
import { View, Text } from 'react-native';
import { Button } from '../../components/Button';
import { Input } from '../../components/StyledComponents';
import styled from '../../utils/styled-components';

export const DisabledButton = () => {
  const [password, setPassword] = useState('');
  const [isPasswordConfirmed, setIsPasswordConfirmed] = useState(false);

  const onConfirm = () => setIsPasswordConfirmed(true);

  return (
    <Container>
      <Input value={password} onChangeText={setPassword} placeholder="password" />
      <Button title="Confirm" onPress={onConfirm} />
      {isPasswordConfirmed && <Text>Password confirmed</Text>}
    </Container>
  );
};

const Container = styled(View)`
  justify-content: space-around;
  flex-grow: 1;
  padding: 16px;
`;
