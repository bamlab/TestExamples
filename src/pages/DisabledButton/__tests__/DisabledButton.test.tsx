import React from 'react';
import { renderPage, getPropsWithNavigation } from '../../../utils/tests/helpers';
import { DisabledButton } from '../DisabledButton';
import { fireEvent } from 'react-native-testing-library';

describe('[Page] DisabledButton', () => {
  const props = getPropsWithNavigation();

  it('shows disabled confirm button while password is blank', () => {
    const page = renderPage(<DisabledButton {...props} />);
    const ConfirmButton = page.getByText('Confirm');
    expect(ConfirmButton).toBeDisabled();
  });

  it('shows success message when password confirmed', () => {
    // Given
    const page = renderPage(<DisabledButton {...props} />);
    const PasswordInput = page.getByPlaceholder('password');
    const ConfirmButton = page.getByText('Confirm');
    // When
    fireEvent.changeText(PasswordInput, 'azertyuiop123');
    expect(ConfirmButton).toBeEnabled();
    fireEvent.press(ConfirmButton);
    // Then
    const SuccessMessage = page.queryByText('Password confirmed');
    expect(SuccessMessage).toBeDefined();
  });
});
