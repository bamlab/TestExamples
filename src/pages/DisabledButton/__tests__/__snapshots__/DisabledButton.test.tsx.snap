// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page] DisabledButton renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexGrow": 1,
        "justifyContent": "space-around",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="password"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00b0ff",
            "borderRadius": 6,
            "color": "white",
            "marginVertical": 2,
            "paddingVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
          ]
        }
      >
        Confirm
      </Text>
    </View>
  </View>
</View>
`;
